src/vcpkg/tools.cpp

```diff
@@ -157,6 +157,30 @@ namespace vcpkg
             });
     }
 
+    ExpectedL<std::string> extract_prefixed_nonquote(StringLiteral prefix,
+                                                     StringLiteral tool_name,
+                                                     std::string&& output,
+                                                     const Path& exe_path)
+    {
+        auto idx = output.find(prefix.data(), 0, prefix.size());
+        if (idx != std::string::npos)
+        {
+            idx += prefix.size();
+            const auto end_idx = output.find_first_of("\"", idx);
+            if (end_idx != std::string::npos)
+            {
+                output.resize(end_idx);
+            }
+
+            output.erase(0, idx);
```

Comment:
Proposed extracted function to also use with the one below:

```
// set target to the subrange [begin_idx, end_idx)
void set_string_to_subrange(std::string& target, size_t begin_idx, size_t end_idx) {
    if (end_idx != std::string::npos)
    {
        output.resize(end_idx);
    }

    output.erase(0, idx);
}
```

The halfopen `[)` notation is not quite accurate since this is indices but I think it's close enough to be understandable.